registry/src/main/java/de/unistuttgart/iaas/amyassist/amy/registry/AbstractTaggableRegistry.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * This source file is part of the Amy open source project.
+ * For more information see github.com/AmyAssist
+ * 
+ * Copyright (c) 2018 the Amy project authors.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information see notice.md
+ */
+
+package de.unistuttgart.iaas.amyassist.amy.registry;
+
+import javax.annotation.Nonnull;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import java.util.List;
+
+/**
+ * A registry with a taggable entity class. Adds the ability to search for entities with a specific tag
+ * @author Benno Krauß
+ * @param <T> entity class
+ */
+public abstract class AbstractTaggableRegistry<T extends RegistryEntity & Taggable> extends AbstractRegistry<T> implements ITaggableRegistry<T> {
+
+    /**
+     * Constants for common applications. Custom tag names are explicitly allowed
+     */
+    public static final String WORK = "work";
```

Comment:
This tags should be defined somewhere in the Location registry, but not here in the generic registry.